### DISCUSSION TITLE
Feature request: Export/Save in PDF

### DISCUSSION THREAD

**Discussion by aamodio** [RussCoder/djvujs#54] [Ideas]
Hi RussCoder,

DjVu file format, unfortunately, is not widely known and supported but It is very efficient. So in my digital project  I will continue to use it but I would like to hide it to users. So an export/save in PDF function can be very useful.

Thanks,
A.

**Comment by RussCoder** [comment]
This is too big a feature. This is definitely out of the scope of this project.
The only not too difficult way how to achieve it, to my mind, is to create a PDF only out of images, without any text layers. Maybe the table of contents can be preserved too. But it won't be a real export to PDF, it would be rather simplified. 

I know there are good libraries to create PDF in the browser, e.g. `pdfmake`. While "DjVu.js" can render all the pages both as `ImageData` or as `URLs to PNG files`. So it's quite achievable. 

But I unlikely will work on it in the near future. I think you can experiment with it yourself, if the simplified solution is enough (however, maybe it's easy to create a text layer via pdfmake too, I don't know). 
And if it works for you, we can then try to integrate it into the viewer.

**Comment by r-pankevicius** [comment]
@aamodio 

While "Save as PDF" is not a feature, but rather it is epic, there is a workaround, though, you can teach your users (I did).

1. Save as djvu (even if it is multipart) as a single file.
2. Have a good native DjVu viewer installed on your machine. I recommend https://windjview.sourceforge.io/ (it is free, GPL, and even has a Mac version).
3. Open saved djvu with that program.
4. Print and select some PDF writer (Adobe PDF, Microsoft Print to PDF) to save as PDF.

That's it.

**Comment by aamodio** [comment]
Thanks, @r-pankevicius  We have already suggested such workaround to our users, but as you know the straight way is the best one. A different approach can be a pass via "Print" command and then use the to print to PDF system extension.

A.
